### ISSUE TITLE
Average and final displacement error metrics

### ISSUE THREAD

**Issue by joseluisvaz** [lyft/l5kit#148]
In `metrics.py`,  `avails` is used to mask out the unvalid errors but then we take the mean over the timesteps
``` python
error = np.sum(((ground_truth - pred) * avails) ** 2, axis=-1)  # reduce coords and use availability
error = error ** 0.5  # calculate root of error (= L2 norm)
error = np.mean(error, axis=-1)  # average over timesteps 
```
Wouldn't it be a better estimate if we used avails to average only wrt the valid timesteps?

I am trying to create some metrics for multi-agent predictions in the cleanest way possible, therefore I am trying to ignore where `avail == 0`

**Comment by lucabergamini** [comment]
Yes, ideally you would want to use `avails` as a mask over the errors. However, if your testing set is fixed using `avails` as a multiplier will just lower (numerically) your final score, so you can still compare different models and have a valid ranking of them. for the competition we were interested in this in particular. I agree we can do better though
